### ISSUE TITLE
resigning a delegate does not remove/clear delegate rank

### ISSUE THREAD

**Issue by dated** [ArkEcosystem/core#3186]
Resigning a delegate should clear the assigned `delegate.rank` attribute, as there will be multiple delegates with the same rank in following rounds otherwise.

This could be done either 

1. directly in the delegate resignation handler, or 
2. in core-state's `buildDelegateRanking` method

**Comment by ArkEcosystemBot** [comment]
**Thanks for opening this issue!** A maintainer will review this in the next few days and explicitly select labels so you know what's going on.

_If no reviewer appears after a week, a reminder will be sent out._

**Comment by dated** [comment]
`2.` looks like the right place for this, so there won't be a gap in the delegate ranks in the round the delegate has resigned in.

**Comment by dated** [comment]
During bootstrap the rank of delegates is not assigned, which begs the next question: what happens with the ranks upon bootstrap during a round a delegate has resigned in?

**Comment by supaiku0** [comment]
@vasild can you take a look?

**Comment by vasild** [comment]
> During bootstrap the rank of resigned delegates is not assigned, which begs the next question: what happens with the ranks upon bootstrap during a round a delegate has resigned in?

During bootstrap and anywhere else the ranks and active delegates for a given round should be calculated without taking into consideration the transactions of that round. I studied the code and I think this is the case, I hope I did not miss something. For example:
```
round 1: blocks 1 .. 51
round 2: blocks 52 .. 102
```
the ranks and active delegates for round 2 should be calculated taking into account all transactions up to and including block 51, but none of the transactions in block 52 and up.

So if we have a resign transaction in block 60, resigning a given delegate that is supposed to forge block 70 in round 2, then he stays as an active delegate in round 2. If he chooses he will be allowed to forge that block 70 or he will miss it if he has shut down operations. For the next round 3 he will be absent from active delegates, irrespective of his vote balance.
